### PR TITLE
APID-208 - Set defaults to addressing fields

### PR DESCRIPTION
**Pull request by sivaprakashiv**

### CODE REVIEW

File: app/uk/gov/hmrc/apiplatformoutboundsoap/models/JsonFormats.scala [hunk 1]
```diff
@@ -16,23 +16,20 @@
...
 
-  private val addressingFrom = getParam("addressing.from")
-  private val addressingReplyTo = getParam("addressing.replyTo")
-  private val addressingFaultTo = getParam("addressing.faultTo")
+  import uk.gov.hmrc.apiplatformoutboundsoap.GlobalContext.injector
+  val appConfig: AppConfig = injector.instanceOf[AppConfig]
 
-  val addressingReads: Reads[Addressing] = (
-    (JsPath \ "from").read[String].orElse(Reads.pure(addressingFrom)) and
+  def addressingReads: Reads[Addressing] = (
```
Comment (worthydolt) [inline app/uk/gov/hmrc/apiplatformoutboundsoap/models/JsonFormats.scala]: It's not clear to me why this has become a `def`

File: app/uk/gov/hmrc/apiplatformoutboundsoap/config/InjectionModule.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+package uk.gov.hmrc.apiplatformoutboundsoap.config
+
+import com.google.inject.AbstractModule
+import uk.gov.hmrc.apiplatformoutboundsoap.GlobalContext
+
+class InjectionModule extends AbstractModule {
+  override def configure() = {
+    // ...
```
Comment (worthydolt) [inline app/uk/gov/hmrc/apiplatformoutboundsoap/config/InjectionModule.scala]: Redundant comment

File: test/uk/gov/hmrc/apiplatformoutboundsoap/services/OutboundServiceSpec.scala [hunk 3]
```diff
@@ -123,8 +126,10 @@ class OutboundServiceSpec extends AnyWordSpec with Matchers with GuiceOneAppPerS
         |<wsa:RelatesTo>RelatesTo</wsa:RelatesTo>""".stripMargin.replaceAll("\n", "")
 
     val mandatoryAddressingHeaders =
```
Comment (worthydolt) [inline test/uk/gov/hmrc/apiplatformoutboundsoap/services/OutboundServiceSpec.scala]: This ``val`` is misnamed (and I know I'm responsible for that). This contains headers that can never be absent but they're _not_ mandatory in the sense of the specification.
Could we call it ``alwaysPresentAddressingHeaders`` or maybe break it into two ``val``s - one called ``mandatoryAddressHeaders`` and one called ``defaultedAddressHeaders``?

File: app/uk/gov/hmrc/apiplatformoutboundsoap/models/MessageRequest.scala [hunk 4]
```diff
@@ -25,13 +25,13 @@ case class  MessageRequest(wsdlUrl: String,
...
                       to: String,
                       replyTo: String,
-                      faultTo: Option[String] = None,
+                      faultTo: String,
                       messageId: String,
                       relatesTo: Option[String] = None){
   validate(to.trim != "", "addressing.to being empty")
   validate(messageId.trim != "", "addressing.messageId being empty")
-  validate(replyTo.trim != "", "addressing.replyTo being empty")
+  validate(from.trim != "", "addressing.from being empty")
```
Comment (worthydolt) [inline app/uk/gov/hmrc/apiplatformoutboundsoap/models/MessageRequest.scala:36]: Are we going to allow callers to supply a ``replyTo`` value that's blank now? Is that what we want?

File: app/uk/gov/hmrc/apiplatformoutboundsoap/models/JsonFormats.scala [hunk 5]
```diff
@@ -36,8 +41,9 @@ object JsonFormats {
     (JsPath \ "wsdlOperation").read[String] and
     (JsPath \ "messageBody").read[String] and
     (JsPath \ "addressing").read[Addressing] and
-    ((JsPath \ "confirmationOfDelivery").read[Boolean] or Reads.pure(false)) and
+    ((JsPath \ "confirmationOfDelivery").read[Boolean].orElse(Reads.pure(appConfig.confirmationOfDelivery)) or Reads.pure(false)) and
```
Comment (worthydolt) [inline app/uk/gov/hmrc/apiplatformoutboundsoap/models/JsonFormats.scala]: We appear to default this to ``false`` in AppConfig so I don't think we need the final ``Reads.pure(false)``

File: app/uk/gov/hmrc/apiplatformoutboundsoap/models/JsonFormats.scala (hunk 1 above)
Comment (sivaprakashiv) [reply app/uk/gov/hmrc/apiplatformoutboundsoap/models/JsonFormats.scala]: haha, its not required. I forgot to change it back to what it is

File: app/uk/gov/hmrc/apiplatformoutboundsoap/models/MessageRequest.scala (hunk 4 above)
Comment (sivaprakashiv) [reply app/uk/gov/hmrc/apiplatformoutboundsoap/models/MessageRequest.scala:36]: In essence if they don't pass any value to replyTo, we will default to empty and also soap spec allows empty value to replyTo. Therefore we allow the caller to pass empty string to keep it consistent

**Review by worthydolt — APPROVED** [review]

**Pull request merged**